libs/hooks/open-telemetry/src/lib/otel-hook.ts

```diff
@@ -0,0 +1,30 @@
+import { FlagMetadata, Logger } from '@openfeature/js-sdk';
+import { Attributes } from '@opentelemetry/api';
+
+export type AttributeMapper = (flagMetadata: FlagMetadata) => Attributes;
+
+export type OpenTelemetryHookOptions = {
+  /**
+   * A function that maps OpenFeature flag metadata values to OpenTelemetry attributes.
+   */
+  attributeMapper?: AttributeMapper;
+};
+
+/**
+ * Base class that does some logging and safely wraps the AttributeMapper.
+ */
+export abstract class OpenTelemetryHook {
```

Comment:
This base class handles wrapping the function to make it safe, since we need to do it in both hooks